### PR TITLE
CORE-9318 Separate DB users for automatic bootstrap and runtime

### PR DESCRIPTION
**Pull request by jacob-scott3141**

### CODE REVIEW

File: charts/corda/templates/_helpers.tpl [hunk 1]
```diff
@@ -561,6 +561,60 @@ Cluster DB credentials environment variables
...
+*/}}
+{{- define "corda.bootstrapClusterDbEnv" -}}
+- name: PGUSER
+  valueFrom:
+    secretKeyRef:
+      {{- if .Values.bootstrap.db.cluster.username.valueFrom.secretKeyRef.name }}
+      name: {{ .Values.bootstrap.db.cluster.username.valueFrom.secretKeyRef.name | quote }}
+      key: {{ required "Must specify bootstrap.db.cluster.username.valueFrom.secretKeyRef.key" .Values.bootstrap.db.cluster.username.valueFrom.secretKeyRef.key | quote }}
+      {{- else }}
+      {{- if .Values.bootstrap.db.cluster.username.value }}
```
Comment (davidcurrie) [inline charts/corda/templates/_helpers.tpl]: nit: I tend to format these as `{{- else if ... }}`

File: charts/corda/templates/bootstrap-cluster-db-secret.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{{- if or ( .Values.bootstrap.db.cluster.password.value) (.Values.bootstrap.db.cluster.username.value) }}
```
Comment (davidcurrie) [inline charts/corda/templates/bootstrap-cluster-db-secret.yaml]: nit: you don't really need the parentheses here given they're only single values - you do have them then I tend to be more generous with the whitespace e.g. ` ( .Values.foo ) `
```suggestion
{{- if or .Values.bootstrap.db.cluster.password.value .Values.bootstrap.db.cluster.username.value }}

```

File: charts/corda/values.yaml [hunk 3]
```diff
@@ -189,6 +189,31 @@ bootstrap:
           key: ""
   # Configuration for database bootstrap
   db:
+    cluster:
```
Comment (davidcurrie) [inline charts/corda/values.yaml:195]: missing a doc comment for `bootstrap.db.cluster`

**Review by davidcurrie — COMMENTED** [review]
A few nits but otherwise LGTM

**Review by yarivus — DISMISSED** [review]
LGTM

**Review by yarivus — DISMISSED** [review]
LGTM

**Review by yarivus — DISMISSED** [review]
LGTM

**Pull request closed without merging**